cmd/get/appcatalogs/runner.go

```diff
@@ -0,0 +1,125 @@
+package appcatalogs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/giantswarm/kubectl-gs/pkg/commonconfig"
+	"github.com/giantswarm/kubectl-gs/pkg/data/domain/app"
+	"github.com/giantswarm/kubectl-gs/pkg/data/domain/appcatalog"
+	"github.com/giantswarm/kubectl-gs/pkg/output"
+)
+
+type runner struct {
+	flag   *flag
+	logger micrologger.Logger
+	fs     afero.Fs
+
+	service appcatalog.Interface
+
+	stdout io.Writer
+	stderr io.Writer
+}
+
+func (r *runner) Run(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
+	err := r.flag.Validate()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.run(ctx, cmd, args)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) error {
+	var err error
+
+	config := commonconfig.New(r.flag.config)
+	{
+		err = r.getService(config)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var name, selector string
+	{
+		if len(args) > 0 {
+			name = strings.ToLower(args[0])
+			selector = fmt.Sprintf("application.giantswarm.io/catalog=%s,latest=true", name)
```

Comment:
This is how we select the appcatalogentries for a catalog. The latest flag is true for the latest version according to semver.